src/adagrams.js

```diff
@@ -1,8 +1,170 @@
+    lettersInHand.push(randomLetters);//
+  }
+  return lettersInHand;
+ },
+
+ usesAvailableLetters(input, lettersInHand) {
+ let inputWord = input.split(""); // split input from user Array ["T", "h", "e"]
+
+ for (let letter of inputWord) { // for every letter in input 
+  let index = lettersInHand.indexOf(letter);// index of where the letter can be found in the array
+  if (index != -1) { //-1 if letter is not present.
+    lettersInHand.splice(index, 1);//removes all the letters if present
+  } 
+  else {
+    return false;
+  }
+}
+return true;
+},
+
+scoreWord(word){ //The  // Wave 3
+  const valueOne = [ "A", "E", "I", "O", "U", "L", "N", "R", "S", "T"	]; //Arrays, things happen inside the function     
+  const valueTwo = ["D", "G"];	     
+  const valueThree = ["B", "C", "M", "P"];	     
+  const valueFour = ["F", "H", "V", "W", "Y"];	     
+  const valueFive = ["K"];	     
+  const valueEight = ["J", "X"];	     
+  const valueTen = ["Q", "Z"];	          
+  let splitWord = word.toUpperCase().split("");   // ["T", "H", "E"] 
+  let wordLength = word.length;
+  let score = 0;     
+  
+  for (let letter of splitWord) {//E  -  score = 6
+    if(valueOne.indexOf(letter)>-1){
+      score = score + 1;
+    }
+    if(valueTwo.indexOf(letter)>-1){
+      score = score + 2;
+    }
+    if(valueThree.indexOf(letter)>-1){
+      score = score + 3;
+    }
+    if(valueFour.indexOf(letter)>-1){
+      score = score + 4;
+    }
+    if(valueFive.indexOf(letter)>-1){
+      score = score + 5;
+    }
+    if(valueEight.indexOf(letter)>-1){
+      score = score + 8;
+    }
+    if(valueTen.indexOf(letter)>-1){
+      score = score + 10;
+    }
+  }
+
+  if(wordLength >= 7 && wordLength <= 10) {
+    score = score + 8;
+  }
+
+  return score;
   },
-};
 
-// Do not remove this line or your tests will break!
-export default Adagrams;
+  highestScoreFrom(words) { //gets an array of strings 
+      let maxScore = 0;
+      let maxRep = 0;
+      let wordMaxScore = "";
+      let list = [];
+
+      for(let word of words) {//build a "hash" with word, score and length 
+        list.push({
+          word:word.toUpperCase(),
+          score: this.scoreWord(word.toUpperCase()),
+          length:word.length
+        });
+      }
+      for(let item of list) { //get max score
+        if(item.score>maxScore){
+          maxScore = item.score;
+        }
+      }
+      for(let item of list) { //how many times does max score repeats
+        if(item.score == maxScore){
+          maxRep = maxRep + 1;//maxRep++;
+          wordMaxScore = item.word;
+        }
+      }
+      if(maxRep === 1) {// if one, we return winner 
+        return {word:wordMaxScore, score:maxScore};
+      }
+      else{ 
+        let wordArray = [];
+          for(let item of list){ //lets get those max
+            if(item.score==maxScore){
+              wordArray.push(item);
```

Comment:
Why not make this list as you're checking to see if there is a second max high score?